botnim/vector_store/vector_store_es.py

```diff
@@ -0,0 +1,144 @@
+                ret = {'id': vs_name, 'name': vs_name}
+        
+        if not ret: # if index does not exist
+            assert not self.init, 'Attempt to create a new vector store after initialization'
+            # Create index with proper mapping
+            mapping = {
+                "mappings": {
+                    "properties": {
+                        "content": {"type": "text"},
+                        "vector": {
+                            "type": "dense_vector",
+                            "dims": 1536,  # OpenAI embedding size
+                            "index": True,
+                            "similarity": "cosine"
+                        }
+                    }
+                }
+            }
+            self.es_client.indices.create(index=vs_name, body=mapping)
+            ret = {'id': vs_name, 'name': vs_name}
+            
+        self.init = True
+        return ret
+
+    def upload_files(self, context, context_name, vector_store, file_streams, callback):
+        count = 0
+        while len(file_streams) > 0:
+            current = file_streams[:32]
+            
+            for filename, content_file, content_type in current:
+                try:
+                    # Read content
+                    content = content_file.read().decode('utf-8')
+                    
+                    # Generate embedding
+                    response = self.openai_client.embeddings.create(
+                        input=content,
+                        model="text-embedding-ada-002",
+                    )
+                    vector = response.data[0].embedding
+
+                    # Index document
+                    self.es_client.index(
+                        index=vector_store['id'],
+                        id=filename,
+                        document={
+                            "content": content,
+                            "vector": vector
+                        }
+                    )
+                except Exception as e:
+                    logger.error(f"Failed to process file {filename}: {str(e)}")
+                    
+            count += len(current)
+            if callable(callback):
+                callback(count)
+            file_streams = file_streams[32:]
+
+    def delete_existing_files(self, context_, vector_store, file_names):
+        try:
+            # Delete documents by their IDs (filenames)
+            body = {
+                "query": {
+                    "terms": {
+                        "_id": file_names
+                    }
+                }
+            }
+            result = self.es_client.delete_by_query(
+                index=vector_store['id'],
+                body=body
+            )
+            return result['deleted']
+        except Exception as e:
+            logger.error(f"Failed to delete files: {str(e)}")
+            return 0
+
+    def update_tools(self, context_, vector_store):
+        if len(self.tools) == 0:
```

Comment:
the tool added needs to be a function, not file-search.

Something like:
```json
{
  "type": "function",
  "name": "search_common_knowledge",
  "description": "Semantic search the 'common-knowledge' vector store",
  "parameters": {
     "type": "object",
     "properties": {
       "query": {
          "type": "string",
          "description": "The query string to use for searching"
       }
     },
     "required": ["query"]
  }
}
```

Reply:
thanks. I did not think of that.